File: 백준/Bronze/1157. 단어 공부/단어 공부.cc
#include <iostream>
#include <string.h>
using namespace std;

int main() {
	string dan;
	int al[26] = { 0, };


	cin >> dan;
	//al에 나온횟수 저장
	for (int i = 0;i < dan.length();i++) {
		if (dan[i] >= 65 && dan[i] <= 90) {
			al[dan[i] - 65]++;
		}
		else if (dan[i] >= 97 && dan[i] <= 122) {
			al[dan[i] - 97]++;
		}
	}

	int max = 0;
	int  m = 0;
	int cnt = 0;



	for (int i = 0;i < 26;i++) {
		if (max > al[i]) {
			continue;
		}
		else if (max < al[i]) {
			max = al[i];
			m = i;
		}
	}
	for (int i = 0;i < 26;i++) {
		if (al[i] == max) {
			cnt = cnt + 1;
		}
	}


	if (cnt > 1) {
		cout << "?";
	}
	else if (cnt == 1) {
		cout << (char)(m + 65);
	}
	
}

File: 백준/Bronze/1157. 단어 공부/단어 공부.py
import sys

cnt = [0 for x in range(28)]

str = sys.stdin.readline().strip()

for i in range(len(str)):
    check = ord(str[i])
    if check>40 and check<91:
        check -= 65
    else:
        check -= 97

    cnt[check] += 1

max = max(cnt)
if cnt.count(max) != 1:
    print('?')
else :
    print(chr(cnt.index(max)+65))